src/mame/drivers/dragon.cpp

```diff
@@ -343,12 +344,25 @@ void dragon64_state::dragon64(machine_config &config)
 	// acia
 	mos6551_device &acia(MOS6551(config, "acia", 0));
 	acia.set_xtal(1.8432_MHz_XTAL);
+	acia.irq_handler().set(FUNC(dragon64_state::acia_irq));
+	acia.txd_handler().set("rs232", FUNC(rs232_port_device::write_txd));
+
+	rs232_port_device &rs232(RS232_PORT(config, "rs232", default_rs232_devices, nullptr));
+	rs232.rxd_handler().set(acia, FUNC(mos6551_device::write_rxd));
+	rs232.dcd_handler().set(acia, FUNC(mos6551_device::write_dcd));
+	rs232.dsr_handler().set(acia, FUNC(mos6551_device::write_dsr));
+	rs232.cts_handler().set(acia, FUNC(mos6551_device::write_cts));
 
 	// software lists
 	SOFTWARE_LIST(config, "dragon_flex_list").set_original("dragon_flex");
 	SOFTWARE_LIST(config, "dragon_os9_list").set_original("dragon_os9");
 }
 
+WRITE_LINE_MEMBER( dragon64_state::acia_irq )
+{
+	m_maincpu->set_input_line(M6809_IRQ_LINE, state ? ASSERT_LINE : CLEAR_LINE);
+}
+
```

Comment:
This is not going to work properly.  You’re setting the IRQ line here without considering that it’s also set by `coco_state::recalculate_irq` which is called by `coco_state::pia0_irq_a` and `coco_state::pia0_irq_b` (connected to the PIA in `dragon_state::dragon_base`).  You need to combine the ACIA IRQ with the PIA IRQs.  The trampolines can probably all be avoided by just using an input merger device.

Reply:
So my current thinking is to override coco_state::irq_get_line in dragon64_state. Then head call the overridden function to logically OR in the ACIA interrupt state.

But the mos6551 device doesn't expose the interrupt state (a hypothetical: m_acia->irq_state()). I thought it would since the real device has an interrupt pin.

If I can't extend mos6551 then I must cache the ACIA interrupt internally in dragon64_state.

Is this acceptable?

Reply:
You should use input mergers.